applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -405,41 +408,39 @@ void KratosGeoSettlement::PrepareModelPart(const Parameters& rSolverSettings)
         domain_part_names.emplace_back(sub_model_part.GetString());
     }
 
-    // Add nodes to computing model part
+    auto collect_ids_from_entity = [](auto& rContainer, auto& rIdSet) {
+        for (const auto& r_item : rContainer) {
+            rIdSet.insert(r_item.Id());
+        }
```

Comment:
This looks like it could be a std::transform:
```suggestion
        std::transform(rContainer.begin(), rContainer.end(), std::inserter(rIdSet, rIdSet.end()), [](const auto& rItem){return r_item.Id();}
```

Reply:
Thanks! Used

Reply:
but reverted for loop in `GenericUtilities::CollectIdsFromEntity` because it works with both vector and set.